src/utils.rs

```diff
@@ -164,30 +165,18 @@ impl<'a, Nalu> NalIterator<'a, Nalu> {
 }
 
 impl<'a> Iterator for NalIterator<'a, H264Nalu<'a>> {
-    type Item = &'a [u8];
+    type Item = Cow<'a, [u8]>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        H264Nalu::next(&mut self.0)
-            .map(|n| {
-                let start = n.sc_offset;
-                let end = n.offset + n.size;
-                &n.data[start..end]
-            })
-            .ok()
```

Comment:
Same here, this should be unrelated to using `Cow` IIUC.

Reply:
Unfortunately it is
```
error[E0515]: cannot return value referencing local data `n.data`
   --> src/utils.rs:175:17
    |
175 |                 &n.data[start..end]
    |                 ^------^^^^^^^^^^^^
    |                 ||
    |                 |`n.data` is borrowed here
    |                 returns a value referencing data owned by the current function

```

I've moved slicing the bitstream to a separate commit, however before that commit this slicing had to be handled separately for `Cow::Owned` , like it is now.